### ISSUE TITLE
GDformat: error with match variable assignment

### ISSUE THREAD

**Issue by NathanLovato** [Scony/godot-gdscript-toolkit#71]
Here's a weird feature supported by match, that causes an error in gdformat:

```gdscript
func _on_Events(msg: Dictionary = {}) -> void:
	match msg:
		{"encounter_icon": var encounter_icon}:
			_encounter.texture = encounter_icon
```

Traceback:

```sh
Traceback (most recent call last):
  File "/home/gdquest/.local/bin/gdformat", line 8, in <module>
    sys.exit(main())
  File "/home/gdquest/.local/lib/python3.7/site-packages/gdtoolkit/formatter/__main__.py", line 49, in main
    comment_parse_tree=comment_parse_tree,
  File "/home/gdquest/.local/lib/python3.7/site-packages/gdtoolkit/formatter/formatter.py", line 52, in format_code
    GLOBAL_SCOPE_SURROUNDING_EMPTY_LINES_TABLE,
  File "/home/gdquest/.local/lib/python3.7/site-packages/gdtoolkit/formatter/block.py", line 39, in format_block
    statement, context
  File "/home/gdquest/.local/lib/python3.7/site-packages/gdtoolkit/formatter/class_statement.py", line 51, in format_class_statement
    return handlers[statement.data](statement, context)
  File "/home/gdquest/.local/lib/python3.7/site-packages/gdtoolkit/formatter/class_statement.py", line 228, in _format_func_statement
    context.create_child_context(last_processed_line_no),
  File "/home/gdquest/.local/lib/python3.7/site-packages/gdtoolkit/formatter/block.py", line 39, in format_block
    statement, context
  File "/home/gdquest/.local/lib/python3.7/site-packages/gdtoolkit/formatter/function_statement.py", line 27, in format_func_statement
    return handlers[statement.data](statement, context)
  File "/home/gdquest/.local/lib/python3.7/site-packages/gdtoolkit/formatter/function_statement.py", line 79, in _format_match_statement
    return _format_branch(prefix, suffix, expr_position, statement, context)
  File "/home/gdquest/.local/lib/python3.7/site-packages/gdtoolkit/formatter/function_statement.py", line 112, in _format_branch
    context.create_child_context(last_processed_line_no),
  File "/home/gdquest/.local/lib/python3.7/site-packages/gdtoolkit/formatter/block.py", line 39, in format_block
    statement, context
  File "/home/gdquest/.local/lib/python3.7/site-packages/gdtoolkit/formatter/function_statement.py", line 27, in format_func_statement
    return handlers[statement.data](statement, context)
  File "/home/gdquest/.local/lib/python3.7/site-packages/gdtoolkit/formatter/function_statement.py", line 86, in _format_match_branch
    return _format_branch(prefix, suffix, expr_position, statement, context)
  File "/home/gdquest/.local/lib/python3.7/site-packages/gdtoolkit/formatter/function_statement.py", line 100, in _format_branch
    expr, expression_context, context
  File "/home/gdquest/.local/lib/python3.7/site-packages/gdtoolkit/formatter/expression.py", line 24, in format_expression
    _format_standalone_expression(concrete_expression, expression_context, context)[
  File "/home/gdquest/.local/lib/python3.7/site-packages/gdtoolkit/formatter/expression.py", line 109, in _format_standalone_expression
    return _format_concrete_expression(expression, expression_context, context)
  File "/home/gdquest/.local/lib/python3.7/site-packages/gdtoolkit/formatter/expression.py", line 116, in _format_concrete_expression
    return _format_foldable(expression, expression_context, context)
  File "/home/gdquest/.local/lib/python3.7/site-packages/gdtoolkit/formatter/expression.py", line 140, in _format_foldable
    single_line_expression = expression_to_str(expression)
  File "/home/gdquest/.local/lib/python3.7/site-packages/gdtoolkit/formatter/expression_to_str.py", line 79, in expression_to_str
    }[expression.data](expression)
KeyError: 'dict_pattern'
```

**Comment by NathanLovato** [comment]
For more details on this kind of code, you can define variables in match blocks, and they take the matched value. It's useful when matching dictionaries:

```gdscript
func _on_Events(msg: Dictionary = {}) -> void:
	match msg:
		{"encounter_icon": var encounter_icon}:
			_encounter.texture = encounter_icon
			not is_open and _animation_player.play("slide_in")
		{"party_member_idx": var party_member_idx, "party_member_icon": var party_member_icon}:
			party_member_idx < _slots.size() and _slots[party_member_idx].setup(party_member_icon)
```

**Comment by Scony** [comment]
Yeah, this is actually not implemented ^^ I'm planning to implement match patterns in the near future.

**Comment by Scony** [comment]
will be implemented in #46
